pkg/controller/atlascluster/atlascluster_controller.go

```diff
@@ -88,15 +95,69 @@ func (r *AtlasClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	return result.ReconcileResult(), nil
 }
 
-func readProjectResource(r *AtlasClusterReconciler, cluster *mdbv1.AtlasCluster, project *mdbv1.AtlasProject) workflow.Result {
+func (r *AtlasClusterReconciler) readProjectResource(cluster *mdbv1.AtlasCluster, project *mdbv1.AtlasProject) workflow.Result {
 	if err := r.Client.Get(context.Background(), cluster.AtlasProjectObjectKey(), project); err != nil {
 		return workflow.Terminate(workflow.Internal, err.Error())
 	}
 	return workflow.OK()
 }
 
 func (r *AtlasClusterReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	filterDeleteOnly := predicate.Funcs{
+		CreateFunc: func(ce event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(ce event.UpdateEvent) bool {
+			return false
+		},
+		GenericFunc: func(ce event.GenericEvent) bool {
+			return false
+		},
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&mdbv1.AtlasCluster{}).
```

Comment:
`For` should be removed if we use `Watches` I guess?

Reply:
That doesn't work for some reason (the whole thing crashes).